test/spec/Feature/Auth/JwtCachingSpec.hs

```diff
@@ -0,0 +1,39 @@
+module Feature.Auth.JwtCachingSpec where
+
+import Network.Wai (Application)
+
+import Network.HTTP.Types
+import Test.Hspec
+import Test.Hspec.Wai
+import Test.Hspec.Wai.JSON
+
+import Protolude  hiding (get)
+import SpecHelper
+
+spec :: SpecWith ((), Application)
+spec = describe "JWT Caching" $ do
+  let auth = authHeaderJWT "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoicG9zdGdyZXN0X3Rlc3RfYXV0aG9yIn0.Xod-F15qsGL0WhdOCr2j3DdKuTw9QJERVgoFD3vGaWA"
+
+  it "jwt Cache warmup" $
+    request methodPost "/rpc/privileged_hello" [auth] [json|{"name": "jdoe"}|]
+      `shouldRespondWith` [json|"Privileged hello to jdoe"|]
+      { matchStatus = 200
+      , matchHeaders = ["Server-Timing" <:> "jwt;dur="
+                       , matchContentTypeJson]
+      }
```

Comment:
>@steve-chavez The caching feature is surely working as shown in local testing above. However the same is not the case when I test it with spec-tests. Maybe there is something missing in spec-tests that I wrote?

Hm, I see on CI that the tests are working fine:

```haskell
  test/spec/Feature/Auth/JwtCachingSpec.hs:19:7: 
  1) Feature.Auth.JwtCachingSpec, JWT Caching, jwt Cache warmup
       missing header:
         Server-Timing: jwt;dur=
       the actual headers were:
         Content-Range: 0-0/*
         Server-Timing: jwt;dur=148.4
         Content-Type: application/json; charset=utf-8
```

See `Server-Timing: jwt;dur=148.4`, but you're not matching the value correctly. You'd only need to check that the value is present with

https://github.com/PostgREST/postgrest/blob/c83befee454bcba29f4e6010f104d260a3058ae0/test/spec/SpecHelper.hs#L49-L53

Reply:
Now I guess you mean why the duration doesn't decrease on the spec tests.

> My guess is that spec-tests resets the AppState after a single request, so that Cache is "reset" after every request.

Yeah, there's no real webserver running in the spec tests. I'd suggest moving these tests to io tests in python.